src/test/java/teammates/test/cases/logic/FeedbackSessionsLogicTest.java

```diff
@@ -2012,163 +1972,6 @@ public void testIsFeedbackSessionFullyCompletedByStudent() throws Exception {
-                assertEquals(String.format(EmailType.FEEDBACK_SESSION_REMINDER.getSubject(), courseName,
-                                           fs.getFeedbackSessionName()),
-                             emailsToInstructor.get(0).getSubject());
-                assertEquals(String.format(EmailType.FEEDBACK_SESSION_REMINDER.getSubject(), courseName,
-                                           fs.getFeedbackSessionName()),
-                             emailsToInstructor.get(1).getSubject());
-            }
-
-        }
-        
-        ______TS("failure: non-existent Feedback session");
-        
-        String nonExistentFsName = "non-ExIsTENT FsnaMe123";
-        
-        try {
-            fsLogic.sendReminderForFeedbackSession(fs.getCourseId(), nonExistentFsName);
-            signalFailureToDetectException();
-        } catch (EntityDoesNotExistException edne) {
-            assertEquals("Trying to remind a non-existent feedback session: "
-                         + fs.getCourseId() + "/" + nonExistentFsName,
-                         edne.getMessage());
-        }
-        
-    }
-    
-    public void testSendReminderForFeedbackSessionParticularUsers() throws Exception {
-        // private method. no need to check for authentication.
-        Logic logic = new Logic();
-        
-        ______TS("typical success case");
-        
-        StudentAttributes studentToRemind = dataBundle.students.get("student5InCourse1");
-        InstructorAttributes instrToRemind = dataBundle.instructors.get("helperOfCourse1");
-        
-        FeedbackSessionAttributes fs = dataBundle.feedbackSessions.get("session1InCourse1");
-        String courseName = CoursesLogic.inst().getCourse(fs.getCourseId()).getName();
-        String[] usersToRemind = new String[] {studentToRemind.email, instrToRemind.email};
-
-        List<EmailWrapper> emailsSent =
-                fsLogic.sendReminderForFeedbackSessionParticularUsers(
-                        fs.getCourseId(), fs.getFeedbackSessionName(), usersToRemind);
-        assertEquals(7, emailsSent.size());
-
-        EmailWrapper emailToStudent = getEmailToStudent(studentToRemind, emailsSent);
-        String errorMessage = "No email sent to selected student " + studentToRemind.email;
-        assertNotNull(errorMessage, emailToStudent);
-        assertEquals(String.format(EmailType.FEEDBACK_SESSION_REMINDER.getSubject(), courseName,
-                                   fs.getFeedbackSessionName()),
-                     emailToStudent.getSubject());
-
-        List<InstructorAttributes> instructorList = logic.getInstructorsForCourse(fs.getCourseId());
-        String notificationHeader = "The email below has been sent to students of course: " + fs.getCourseId();
-        for (InstructorAttributes i : instructorList) {
-            List<EmailWrapper> emailsToInstructor = getEmailsToInstructor(i, emailsSent);
-            
-            if (i.email.equals(instrToRemind.email)) {
```

Comment:
@kanghj my answer here is common for both your concerns, only differing in where the actions are moved to.

You would have noticed that the actions removed here are all task-queue-adding actions, i.e they add stuffs to task queue. Most task-queue-adding actions were not testable before (fixing which is the very point of this PR), and for email-queueing actions like these, they were previously tested by returning the `List` of `EmailWrapper`s being *generated for queueing* (notice how I didn't say being *queued*). This is not the most accurate way of testing whether the emails are *actually in the queue*, but a good enough way considering the circumstances.

Now that the queueing action is fully testable, the queue itself can be examined for the tasks added, and these three actions are no exception (no action is an exception). All of them have tests written in the other PR #6276: `FeedbackSessionRemindEmailWorkerActionTest`, `FeedbackSessionRemindParticularUsersEmailWorkerActionTest`, and `InstructorCourseRemindActionTest`, all testing whether the tasks are queued correctly.

Having said that, the level of detail of the new tests is not matching this one yet. The new tests I've written there only test the *names* of the tasks being queued, not *the tasks* themselves (e.g. it can only test that an email is being queued, but not what email and to whom the email is sent etc.). This is because `MockTaskQueuer` is still at its youngest stage, otherwise that other PR will be a worse nightmare to review :p Improving the infrastructure of `MockTaskQueuer` will definitely make testing at the same detail (and even greater) possible, but this is for another time.

So, to answer your question: no, or at least not yet, but easily doable (albeit a dirty work).